keel-sql/src/main/kotlin/com/netflix/spinnaker/keel/sql/SqlArtifactRepository.kt

```diff
@@ -565,53 +592,31 @@ class SqlArtifactRepository(
       return false
     }
 
-    return jooq
-      .select(
-        ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_STATUS,
-        ENVIRONMENT_ARTIFACT_VERSIONS.PROMOTION_REFERENCE)
-      .from(ENVIRONMENT_ARTIFACT_VERSIONS)
-      .where(
-        ENVIRONMENT_ARTIFACT_VERSIONS.ENVIRONMENT_UID.eq(envUid),
-        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_UID.eq(artUid),
-        ENVIRONMENT_ARTIFACT_VERSIONS.ARTIFACT_VERSION.eq(veto.version))
-      .fetchOne { (_, reference) ->
-        if (reference != null && !force) {
-          log.warn(
-            "Not vetoing artifact version as it appears to have already been an automated rollback target: " +
-              "deliveryConfig=${deliveryConfig.name}, " +
-              "environment=${veto.targetEnvironment}, " +
-              "artifactVersion=${veto.version}, " +
-              "priorVersionReference=$reference")
-          return@fetchOne false
+    return selectArtifactStatusAndPromotionReferenceInTargetEnvironment(envUid, artUid, veto.version)
+      .fetchOne { (_, ref) ->
```

Comment:
if we're not using status (as `(_,ref)` indicates), can we just not fetch it?

Reply:
The embarrassing truth is that I tried, but I couldn't get it to compile. Because `fetchOne` is an overloaded function, if you pass in only one argument, the Kotlin compiler doesn't know which method to use, so you need to disambguate it with explicit type information. I couldn't figure out the right incantation to get it to work. I may give it another shot.

Reply:
It's working now. :shrug:

Fixed in 946564cce

Reply:
Ugh, so it compiles with this change but it breaks the tests. If I can't figure. out why, I'll probably just revert to fetching the status and not using it.